fsspec/implementations/dropboxdrivefs.py

```diff
@@ -0,0 +1,244 @@
+        super().__init__(**storage_options)
+        self.token = storage_options["token"]
+        self.kwargs = storage_options
+        self.connect()
+
+    def connect(self):
+        """ connect to the dropbox account with the given token
+        """
+        self.dbx = dropbox.Dropbox(self.token)
+        self.session = requests.Session()
+        self.session.auth = ("Authorization", self.token)
+
+    def ls(self, path, detail=True, **kwargs):
+        """ List objects at path
+        """
+        while "//" in path:
+            path = path.replace("//", "/")
+        list_file = []
+        list_item = self.dbx.files_list_folder(
+            path, recursive=True, include_media_info=True
+        )
+        items = list_item.entries
+        while list_item.has_more:
+            list_item = self.dbx.files_list_folder_continue(list_item.cursor)
+            items = list_item.entries + items
+
+        if detail:
+            for item in list_item.entries:
+                if isinstance(item, dropbox.files.FileMetadata):
+                    list_file.append(
+                        {"name": item.path_display, "size": item.size, "type": "file"}
+                    )
+                elif isinstance(item, dropbox.files.FolderMetadata):
+                    list_file.append(
+                        {"name": item.path_display, "size": None, "type": "folder"}
+                    )
+                else:
+                    list_file.append(
+                        {"name": item.path_display, "size": item.size, "type": "unknow"}
+                    )
+        else:
+            for item in list_item.entries:
+                list_file.append(item.path_display)
+        return list_file
+
+    def _open(
+        self,
+        path,
+        mode="rb",
+        block_size=None,
+        autocommit=None,
+        cache_options=None,
+        **kwargs
+    ):
+        return DropboxDriveFile(
+            self,
+            self.dbx,
+            path,
+            session=self.session,
+            block_size=4 * 1024 * 1024,
+            mode=mode,
+            **kwargs
+        )
+
+    def info(self, url, **kwargs):
+        """Get info of URL
+        """
+        metadata = self.dbx.files_get_metadata(url)
+        if isinstance(metadata, dropbox.files.FileMetadata):
+            return {
+                "name": metadata.path_display,
+                "size": metadata.size,
+                "type": "file",
+            }
+        elif isinstance(metadata, dropbox.files.FolderMetadata):
+            return {"name": metadata.path_display, "size": None, "type": "folder"}
+        else:
+            return {"name": url, "size": None, "type": "unknow"}
+
+
+class DropboxDriveFile(AbstractBufferedFile):
+    """ fetch_all, fetch_range, and read method are based from the http implementation
+    """
+
+    def __init__(
+        self, fs, dbx, path, session=None, block_size=None, mode="rb", **kwargs
```

Comment:
I'd recommend matching the signature of the base class. 
```suggestion
        self,
        fs,
        path,
        mode="rb",
        block_size="default",
        autocommit=True,
        cache_type="readahead",
        cache_options=None,
        *,
        dbx
        **kwargs
```

That uses a keyword only argument for dbx. Though I wonder: can you just get it from `fs` instead?